rust/rbac-registration/src/registration/cardano/mod.rs

```diff
@@ -56,9 +59,14 @@ impl RegistrationChain {
     /// # Errors
     ///
     /// Returns an error if data is invalid
-    pub fn update(&self, cip509: Cip509) -> anyhow::Result<Self> {
-        let new_inner = self.inner.update(cip509)?;
-
+    pub fn update(&self, cip509: Cip509, raw_aux_data: &[u8]) -> anyhow::Result<Self> {
+        let latest_signing_pk = self.get_latest_signing_pk_for_role(&RoleId::Role0);
+        let new_inner = match latest_signing_pk {
+            Some((signing_pk, _)) => self.inner.update(cip509, signing_pk, raw_aux_data)?,
+            None => {
+                bail!("No latest signing key found for role 0, cannot perform signature validation")
```

Comment:
Perhaps it would be convenient to update the problem report of the given registration. As far as I understand, we will eventually have something like that in indexing:
```rust
let cip509 = Cip509::new(...)?;
match rbac_cache.add(cip509) {
    Ok(...) => {
        // All good: the cache is updated and we can add the registration to database.
       <insert into rbac_registration table>
    },
    Err(...) => {
        <insert into rbac_invalid_registration table>
    }
}
```
When we inserting an entry into the `rbac_invalid_registration`, we also storing the problem report of that registration.